### PR TITLE
chore(external features): deleted all relevant code for external feature

### PR DESCRIPTION
**Pull request by thehulke**
Deleted all code relevant for external feature, it's basically dead code.
When we will get to implement it, we will overhaul `engine-core` so it will be able to run features mid-flight.

Everything that was deleted in this PR is irrelevant for now.

### CODE REVIEW

File: packages/scripts/src/application.ts [hunk 1]
```diff
@@ -228,8 +187,6 @@ export class Application {
             singleFeature,
             // should build this feature in external mode
             isExternal: external,
```
Comment (cylon3035) [inline packages/scripts/src/application.ts:189]: this flag as well

**Comment by barak007** [comment]
In the end we need to go all the way around...all features should be external. I can see a way to achieve this is by first delete everything first but I would want to see what's going to replace it. 
The minimal working parts are:
1. allow to load features in the entry point (even from url param) 
2. document how to build and maintain features complaint with an already built application (externals in general)

If we only remove the support for "external features" to integrate with other node environments. this become a much easier task. 
then we could re-think the node/processes integration

**Review by barak007 — COMMENTED** [review]
In the end we need to go all the way around...all features should be external. I can see a way to achieve this is by first delete everything first but I would want to see what's going to replace it. 
The minimal working parts are:
1. allow to load features in the entry point (even from url param) 
2. document how to build and maintain features complaint with an already built application (externals in general)

If we only remove the support for "external features" to integrate with other node environments. this become a much easier task. 
then we could re-think the node/processes integration 


About the PR itself I would recommend to review it in IDE sine we are removing stuff other related things might not come up in the diff

**Review by AviVahl — APPROVED** [review]
merge/rebase master to this branch and land this.

File: packages/engineer/src/cli-commands.ts [hunk 2]
```diff
@@ -142,8 +142,6 @@ export function buildCommand(program: Command) {
         .option('--outDir <outDir>', 'output directory for the built application', 'dist-app')
         .option('--webpackConfig <webpackConfig>', 'path to webpack config to build the application with')
         .option('--publicConfigsRoute <publicConfigsRoute>', 'public route for configurations')
-        .option('--external [true|false]', 'build feature as external', parseBoolean, false)
-        .option('--eagerEntrypoints [true|false]', 'build feature as external', parseBoolean, false)
```
Comment (AviVahl) [inline packages/engineer/src/cli-commands.ts:146]: this has nothing to do with external. it has to do with how the dynamic import is marked (for the entry point engine-scripts creates). see 'eager' @ https://webpack.js.org/api/module-methods/#magic-comments

could have been kept, but not crucial

File: packages/electron-commons/test/initializers/electron-node.spec.ts [hunk 3]
```diff
@@ -20,17 +20,15 @@ const setupRunningEnv = async ({
...
         env: serverEnv,
-        getApplicationMetaData: () =>
-            Promise.resolve({
-                featureName: testFeature.id,
-                basePath: testProjectPath,
-                outputPath: fs.join(testProjectPath, 'dist-app'),
-                nodeEntryPath: fs.join(testProjectPath, 'entry.js'),
-                config: [testFeature.use({ errorType: { type: errorMode, handleUncaught } })],
-                features: Array.from(features.entries()),
-                env: serverEnv,
```
Comment (AviVahl) [inline packages/electron-commons/test/initializers/electron-node.spec.ts]: is env not needed anymore?

File: packages/electron-node/src/initializers/browser-window.ts [hunk 4]
```diff
@@ -1,8 +1,8 @@
+import { once } from '@wixc3/common';
```
Comment (AviVahl) [inline packages/electron-node/src/initializers/browser-window.ts:1]: change unrelated to PR? could have been done in a separate one

File: packages/engineer/src/feature/dev-server.dev-server.env.ts [hunk 5]
```diff
@@ -1,26 +1,23 @@
-import type io from 'socket.io';
-import devServerFeature, { devServerEnv } from './dev-server.feature';
-import { TargetApplication } from '../application-proxy-service';
-import express from 'express';
+import { createDisposables } from '@wixc3/create-disposables';
```
Comment (AviVahl) [inline packages/engineer/src/feature/dev-server.dev-server.env.ts:1]: the imports reordering makes the review of the actual changes much harder.

File: packages/scripts/src/application/application.ts [hunk 6]
```diff
@@ -76,15 +61,9 @@ export class Application {
...
         const opts = defaults(options, buildDefaults);
-        const { config: _config, path: configPath } = await this.getEngineConfig();
-        const config: EngineConfig = defaults(_config, { externalFeatureDefinitions: [] as IExternalDefinition[] });
+        const { config: _config } = await this.getEngineConfig();
+        const config = defaults(_config, buildDefaults);
 
-        if (opts.external && !opts.featureName) {
-            throw new Error('You must specify a feature name when building a feature in external mode');
-        }
-        if (config.require) {
```
Comment (AviVahl) [inline packages/scripts/src/application/application.ts:85]: this is important. engine.config.js -> require are hooks that we use to install `@stylable/node/r` and others

File: packages/scripts/src/application/application.ts [hunk 7]
```diff
@@ -93,104 +72,24 @@ export class Application {
...
-
-        if (opts.includeExternalFeatures && config.externalFeatureDefinitions.length) {
-            externalFeatures.push(
-                ...getExternalFeaturesMetadata(config.externalFeatureDefinitions, externalFeaturesBasePath)
-            );
-        }
-
-        // creating external-features json either way
-        this.writeExtFeaturesJson(opts, externalFeatures);
+        const sourceRoot = opts.sourcesRoot ?? opts.featureDiscoveryRoot ?? '.';
```
Comment (AviVahl) [inline packages/scripts/src/application/application.ts:78]: rename opts into something meaningful

File: packages/scripts/src/application/defaults.ts [hunk 8]
```diff
@@ -1,13 +1,8 @@
 import type { IBuildCommandOptions } from './types';
 
-export const DEFAULT_EXTERNAL_FEATURES_PATH = 'external-features.json';
-
 export const buildDefaults = {
     mode: 'production',
```
Comment (AviVahl) [inline packages/scripts/src/application/defaults.ts:4]: not sure keeping this file now improves the readability, considering most code was removed

**Review by AviVahl — APPROVED** [review]

File: packages/engineer/src/cli-commands.ts (hunk 2 above)
Comment (thehulke) [reply packages/engineer/src/cli-commands.ts:146]: IMO It's best to cleanup at this case, and re-implement if needed

**Pull request merged**